backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/log/ModularContractLogService.java

```diff
@@ -392,7 +397,13 @@ public void unprocessModularContractLogs(@NonNull final InvoiceId invoiceId, @No
 		 InterfaceWrapperHelper.saveAll(invoiceCandidates);
 		 if (docTypeBL.isDefinitiveInvoiceOrDefinitiveCreditMemo(docTypeId))
 		 {
-			 flatrateBL.reverseDefinitiveInvoice(ImmutableSet.of(Check.assumePresent(flatrateBL.getIdByInvoiceId(invoiceId), "FlatrateTermId should be present")));
+			 final Set<FlatrateTermId> contractIds = invoiceBL.getLines(invoiceId)
+					 .stream()
+					 .map(org.compiere.model.I_C_InvoiceLine::getC_Flatrate_Term_ID)
+					 .map(FlatrateTermId::ofRepoIdOrNull) // invoice might have been imported from a csv-file and have no flatrate term 
+					 .filter(Objects::nonNull)
+					 .collect(Collectors.toSet());
+			 flatrateBL.reverseDefinitiveInvoice(contractIds);
```

Comment:
Why drop `flatrateBL.getIdByInvoiceId(invoiceId)` which does the same and also ensures that we don't have multiple Ids in the same Invoice?

Reply:
nvm, `de.metas.util.collections.CollectionUtils#extractDistinctElements` excludes nulls, which I didn't notice